### DISCUSSION TITLE
Failed to connect to ESP8266: Timed out waiting for packer header

### DISCUSSION THREAD

**Discussion by wouter-jacobs** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#310] [Q&A]
When I try to upload the sketch I get the following error in the terminal: Failed to connect to ESP8266: Timed out waiting for packer header. Does anyone have an idea to solve this? Been trying for hour
![image 2](https://user-images.githubusercontent.com/36419563/175958091-9f3e04be-c383-403e-b2a9-8d802415c97d.jpeg)
![image 1](https://user-images.githubusercontent.com/36419563/175958100-ea620207-fbb0-454f-a363-5619bb8efab3.jpeg)

**Comment by visualapproach** [comment]
Do you have a data USB cable, as opposed to a charging cable?
I had some issues with platformio update today but a restart everything in da house solved it.

**Reply by wouter-jacobs** [reply]
Thanks for the tip. I do use a USB data cable. I think the ESP8266 won't go into flash mode. I've tried several tips but without success so far.

**Reply by visualapproach** [reply]
What if you press and hold flash button. Press reset button. Upload from computer. When the squiggly line is coming up, release flash button

**Reply by wouter-jacobs** [reply]
This also doesn't seem to work. Tried it on a laptop just to be sure, but still got the same error.

**Reply by visualapproach** [reply]
Okay, I'll see if I can do it. Is it only the filesystem that fails or is it the sketch also?

**Reply by wouter-jacobs** [reply]
the error message comes with both filesystem and sketch. I also tried it with my laptop. I also get the same error here. Maybe made a mistake or damage with the soldering?🙈

**Reply by visualapproach** [reply]
If you want you can post a pic on the PCB back side

**Reply by wouter-jacobs** [reply]
![image](https://user-images.githubusercontent.com/36419563/176020151-3a602fef-f35a-4faa-bd16-7254375206b8.jpeg)
![image](https://user-images.githubusercontent.com/36419563/176020203-bb8af36a-0984-49d6-b91d-5a2205f4dccd.jpeg)
![image](https://user-images.githubusercontent.com/36419563/176020244-bbb01142-69fb-4d85-b5ae-ec7fd4b416ba.jpeg)

**Reply by visualapproach** [reply]
Don't see any obvious shorts. So shouldn't be it. And unless you really toasted it I don't think you destroyed the ESP either.
Do you have another ESP to try stand alone without the PCB? Or use Arduino IDE to just upload anything to see if it's the ESP or VS that is not working.

**Reply by wouter-jacobs** [reply]
first of all thanks for taking the time to help. just ordered an extra ESP. This will be delivered tomorrow. Will test tomorrow night with the new separate ESP. If this works, can I solder it to the pcb after uploading the sketch? or will this cause problems?

**Reply by visualapproach** [reply]
No problem. Yes that's a good way to minimize the potential failure points. And if you can, use female headers instead of soldering directly to the board. Much easier to troubleshoot and fix things :)

**Reply by wouter-jacobs** [reply]
Just ordered the female headers. I should have done right away. Tomorrow I will give an update. Good evening!

**Comment by visualapproach** [comment]
Morning, long shot, but maybe changing the baudrate will do the trick. Find "platformio.ini" and comment out the "upload_speed = 921600" parameter by putting a semicolon in front. Ctrl+s to save. Compile and upload.

**Reply by wouter-jacobs** [reply]
Thanks for the tip. still the same error. Just tried to connect to Arduino IDE. Here I get the same error.

esptool.FatalError: Failed to connect to ESP8266: Timed out waiting for packet header

I think this allows us to conclude that the problem is in the ESP?

**Reply by visualapproach** [reply]
Yes I'd think so. I discovered that the host name "layzspa.local" didn't work on one of my computers when uploading over the air, so I had to put the ip number instead but that's not your problem since you use usb.

**Comment by wouter-jacobs** [comment]
Just unpacked the new nodeMCU and handed it to the computer. And with results. The filesystem image upload went well. Now I find out that I ordered a SP8266 12 F instead of the SP8266 12 E. Is this a problem? When I google this I read that the 12F has a better optimized antenna but is actually equal to the 12E version..
![upload Filesystem](https://user-images.githubusercontent.com/36419563/176251050-d0434cbf-bce3-42c7-a5ba-8e6a26b8d637.PNG)

**Reply by visualapproach** [reply]
should not be a problem as long as it fits physically in the pcb. Great that it works now.

**Reply by wouter-jacobs** [reply]
It seems to work. There is a wifi network online called Lay-Z-Spa module. For this I have to enter a password. Does this match the manual? I was expecting an access point called "Auto portal"..![image](https://user-images.githubusercontent.com/36419563/176252770-e5704707-0357-4201-91a8-d581f0dd95f9.jpeg)

**Reply by visualapproach** [reply]
I don't remember but the password should be the same. Otherwise check out the config.h file. (Not at computer now)

**Comment by wouter-jacobs** [comment]
Good morning, now put the PCB in the pump and it has been running for 24 hours. Everything seemed to be going well until this morning. The lights on the display were still on, but I could no longer operate them. After a reset of both the pump and the ESP, I no longer get any lights on the display. The ESP does come online and I can turn on the pump / heater, but it turns off after 5 seconds.

**Reply by visualapproach** [reply]
Strange. Try the 2022-06-29 version in dev branch if you haven't.

**Reply by wouter-jacobs** [reply]
the version I have installed now is the 24-5 So I have to put a totally new sletch on the ESP via usb or could it upload via web module?![image](https://user-images.githubusercontent.com/36419563/176861224-1b46a77d-6bd1-479f-8ece-ecbcff13d709.jpeg)

**Reply by visualapproach** [reply]
You can do it OTA or as before with USB. See build instruction for details. USB is more reliable but can be inconvenient

**Reply by wouter-jacobs** [reply]
maybe a stupid question but I only see the code from 28-05 as the last release. Do I need to change a setting somewhere to enable the dev. get version?

**Reply by visualapproach** [reply]
Select branch on the GitHub page from master to development, or if you synced vs code you can do it there too

**Reply by wouter-jacobs** [reply]
Thanks for the tip the dev. version found!

I'm going to try the new version and will report if this solution works for me.

One last question.. do I have to delete the previous version first or will it be overwritten if I use this Dev. version on the ESP via USB

**Reply by visualapproach** [reply]
It will overwrite. Upload new html files also if they are newer. Via web ui 'upload.html'

**Reply by wouter-jacobs** [reply]
The new DEV version is on the ESP. Had the pcb in the pump. The pump can now be operated via remote control. Only the display of the pump remains off. If I remove the pcb the screen works. Do I still need to make an adjustment to my PCB? Strange that it worked for 24 hours...

**Reply by visualapproach** [reply]
Check your solder joints and cables

**Comment by wouter-jacobs** [comment]
Good morning, yesterday I assembled the new module. A new installation on the NodeMCU (dev version 29-06) with a newly soldered TXS0108E level converter. The pump starts up and operates via web module and display on the pump. After a few hours, the pump jammed again. Just like a week ago. Then with a different software version and a different NodeMCU. The circulation pump is still running and the LED lights on the pump are on. The control panel does not respond and the web module no longer works. No idea what to try now...

**Reply by visualapproach** [reply]
Morning! The fact that it works a couple of hours makes me think it must be something physical that changes. Poor solders or wires that just makes contact initially may disconnect due to vibrations, thermal expansion, moisture etc.

**Comment by wouter-jacobs** [comment]
Thanks for your response.   I will prepare a second PCB this coming weekend. luckily I now work with the female header pins. Hope this new PCB solves the problem.

**Comment by wouter-jacobs** [comment]
Just remember that I used a new pcb board after all. The only thing I have reused are the wires with connector. Could a cable that is too long be the cause of the jamming?

**Reply by visualapproach** [reply]
Or a broken cable...
